lib/Doctrine/ruleset.xml

```diff
@@ -334,17 +334,40 @@
     <!-- Require types to be written as natively if possible;
          require iterable types to specify phpDoc with their content;
          forbid useless/duplicated information in phpDoc -->
-    <rule ref="SlevomatCodingStandard.TypeHints.TypeHintDeclaration">
+    <rule ref="SlevomatCodingStandard.TypeHints.ParameterTypeHint">
+        <properties>
+            <property name="traversableTypeHints" type="array">
+                <element value="Doctrine\Common\Collections\Collection"/>
```

Comment:
It isn't fully clear to me why we are targeting `Collection` as a specific type here.

Would need:

 1. an example (before/after test)
 2. to know how this copes with `@var Collection<TKey, TVar>`

Reply:
> It isn't fully clear to me why we are targeting `Collection` as a specific type here.

Same reason why all other sniffs of this kind add this: to ensure that a parameter with a `Collection` type hint also gets a proper `@param` or `@return` annotation that clarifies its type (as is done with `array`).

+1 on adding tests for this.

Reply:
Yeah, but shouldn't this be done with any `Traversable` then?

Reply:
It was migrated from deprecated sniff.

According to docs, this forces to write `\Doctrine\Common\Collections\Collection|Foo[]`.

Personally, I always write `\Doctrine\Common\Collections\Collection<Foo>`. If it was on me, I'd throw it away.

Reply:
`array` and `iterable` are always treated as such. This list is in addition to the ones where the sniff requires it by default.

Reply:
So we keep it and need:

- [ ]  + test
- [ ]  change type to `Traversable` instead

Reply:
@kukulich do we actually need `Traversable` here? I thought that was covered by the root sniff already?

Reply:
@alcaeus by default it seems to be empty https://github.com/slevomat/coding-standard/blob/7a9530a05f7b93cd3cdb8c475f743f97a447aae4/SlevomatCodingStandard/Sniffs/TypeHints/ParameterTypeHintSniff.php#L343-L351

Reply:
@alcaeus Only `iterable` and `array` are covered by default. You can use [DisallowArrayTypeHintSyntaxSniff](https://github.com/slevomat/coding-standard/blob/master/SlevomatCodingStandard/Sniffs/TypeHints/DisallowArrayTypeHintSyntaxSniff.php) and use always generic.

Reply:
I really like the idea of enabling `DisallowArrayTypeHintSyntaxSniff`

WDY-all-T?

Reply:
Please create a separate PR for that so the deciders can vote on it. Thanks!

Reply:
If I add only Traversable there it is not applied to eg. Iterator. So we have to explicitly name everything, that implements Traversable